lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -0,0 +1,602 @@
+                self._b_working_directory,
+            )
+        elif collection.is_file or collection.is_dir or collection.is_subdirs:
+            b_artifact_path = to_bytes(collection.src)
+        else:
+            # NOTE: This may happen `if collection.is_online_index_pointer`
+            raise RuntimeError(
+                'The artifact is of an unexpected type {art_type!s}'.
+                format(art_type=collection.type)
+            )
+
+        self._artifact_cache[collection.src] = b_artifact_path
+        return b_artifact_path
+
+    def _get_direct_collection_namespace(self, collection):
+        # type: (Candidate) -> Optional[str]
+        assert collection.is_concrete_artifact
+        return self.get_direct_collection_meta(collection)['namespace']  # type: ignore
+
+    def _get_direct_collection_name(self, collection):
+        # type: (Candidate) -> Optional[str]
+        assert collection.is_concrete_artifact
+        return self.get_direct_collection_meta(collection)['name']  # type: ignore
+
+    def get_direct_collection_fqcn(self, collection):
+        # type: (Candidate) -> Optional[str]
+        assert collection.is_concrete_artifact
+
+        if collection.is_virtual:
+            # NOTE: should it be something like "<virtual>"?
+            return None
+
+        return '.'.join((  # type: ignore
+            self._get_direct_collection_namespace(collection),  # type: ignore
+            self._get_direct_collection_name(collection),
+        ))
+
+    def get_direct_collection_version(self, collection):
+        # type: (Union[Candidate, Requirement]) -> str
+        assert collection.is_concrete_artifact
+        return self.get_direct_collection_meta(collection)['version']  # type: ignore
+
+    def get_direct_collection_dependencies(self, collection):
+        # type: (Union[Candidate, Requirement]) -> Dict[str, str]
+        assert collection.is_concrete_artifact
+        return self.get_direct_collection_meta(collection)['dependencies']  # type: ignore
+
+    def get_direct_collection_meta(self, collection):
+        # type: (Union[Candidate, Requirement]) -> Dict[str, Optional[Union[str, Dict[str, str], List[str]]]]
+        assert collection.is_concrete_artifact
+
+        try:  # FIXME: use unique collection identifier as a cache key?
+            return self._artifact_meta_cache[collection.src]
+        except KeyError:
+            b_artifact_path = self.get_artifact_path(collection)
+
+        if collection.is_url or collection.is_file:
+            collection_meta = _get_meta_from_tar(b_artifact_path)
+        elif collection.is_dir:  # should we just build a coll instead?
+            # FIXME: what if there's subdirs?
+            try:
+                collection_meta = _get_meta_from_dir(b_artifact_path)
+            except LookupError as lookup_err:
+                raise_from(
+                    AnsibleError(
+                        'Failed to find the collection dir deps: {err!s}'.
```

Comment:
If I have a collection in my install path that doesn't have a MANIFEST.json or galaxy.yml and try to install an unrelated collection, I get `ERROR! Failed to find the collection dir deps: The collection galaxy.yml path '/home/fedora/ansible/test_collections/ansible_collections/test/coll/galaxy.yml' does not exist.` We could probably just determine the namespace and name from the path, and the version would be "*" and no dependencies.

Reply:
Could you add an integration test for that in a PR so I could attempt to see what state it is in, in runtime?